src/Engine/Renderer/Mesh/Mesh.inl

```diff
@@ -387,13 +387,13 @@ IndexedGeometry<T>::IndexedGeometry( const std::string& name,
 
 template <typename T>
 void IndexedGeometry<T>::loadGeometry( T&& mesh ) {
+    if ( base::m_mesh.m_indices != mesh.m_indices ) { m_indicesDirty = true; }
```

Comment:
This could be extremely costly to compare the contents of two vectors. just think about comparing the indices of two triangle meshes with 500k triangles (medium complexity) - 1.5M of comparisons if indices are the same ... ? What about an other mean, very less costly, to set indices dirty (a topologyChanged method ?). @dlyr What do you think of that ?

Reply:
Well, we will discuss that point AFK.
But the loadGeometry do change topology since it changes the associated CoreGeometry.

I think we don't have to test for equality, loadGeometry should be only called to move another CoreGeometry to the Render::Mesh (and release the current one).
If you want to update attributes, use getCoreGeometry() and modify the attributes.

Right way to do is to have similar observer thing on indices, small patch could be to add a setDirty to the VaoIndices class.

Reply:
I agree on the usage restriction of loadGeometry. And this restriction must be mentioned in the method documentation.
It could be helpful also to add documentation about this use case as it is quite current.